addons/website_event/static/src/snippets/s_events/000.js

```diff
@@ -0,0 +1,30 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+import DynamicSnippet from 'website.s_dynamic_snippet';
+
+const DynamicSnippetEvents = DynamicSnippet.extend({
+    selector: '.s_event_upcoming_snippet',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     * @private
+     */
+    _getSearchDomain: function () {
+        const searchDomain = this._super.apply(this, arguments);
+        const filterByTagIds = JSON.parse(this.$el.get(0).dataset.filterByTagIds || '[]');
+        if (filterByTagIds.length > 0) {
+            for (let i = 1; i < filterByTagIds.length; i++) {
+                searchDomain.push('&');
+            }
```

Comment:
```suggestion
            searchDomain.concat(Array(filterByTagIds.length-1).fill('&');
```

But `&` or `|` ? :thinking:

Reply:
I'm staying with the `&` since all the selected tags must be present :)
I fixed the selected tag(s) after editing the snippet. Thanks for pointing the bug out !